gtest_catch_exceptions_test fails with gcc-6 / g++-6

```
/usr/bin/ctest --force-new-ctest-process -j12
Test project /<<PKGBUILDDIR>>/obj-x86_64-linux-gnu
      Start  1: gtest-death-test_test
      Start  2: gtest_environment_test
      Start  3: gtest-filepath_test
      Start  4: gtest-linked_ptr_test
      Start  5: gtest-listener_test
      Start  6: gtest_main_unittest
      Start  7: gtest-message_test
      Start  8: gtest_no_test_unittest
      Start  9: gtest-options_test
      Start 10: gtest-param-test_test
      Start 11: gtest-port_test
      Start 12: gtest_pred_impl_unittest
 1/41 Test  #4: gtest-linked_ptr_test ..............   Passed    0.01 sec
 2/41 Test  #5: gtest-listener_test ................   Passed    0.01 sec
 3/41 Test  #6: gtest_main_unittest ................   Passed    0.10 sec
 4/41 Test  #7: gtest-message_test .................   Passed    0.10 sec
 5/41 Test  #8: gtest_no_test_unittest .............   Passed    0.10 sec
 6/41 Test #10: gtest-param-test_test ..............   Passed    0.14 sec
 7/41 Test #12: gtest_pred_impl_unittest ...........   Passed    0.14 sec
 8/41 Test  #9: gtest-options_test .................   Passed    0.14 sec
 9/41 Test  #3: gtest-filepath_test ................   Passed    0.24 sec
10/41 Test  #2: gtest_environment_test .............   Passed    0.24 sec
      Start 13: gtest_premature_exit_test
      Start 14: gtest-printers_test
      Start 15: gtest_prod_test
      Start 16: gtest_repeat_test
      Start 17: gtest_sole_header_test
      Start 18: gtest_stress_test
      Start 19: gtest-test-part_test
      Start 20: gtest_throw_on_failure_ex_test
      Start 21: gtest-typed-test_test
      Start 22: gtest_unittest
11/41 Test #15: gtest_prod_test ....................   Passed    0.02 sec
12/41 Test #17: gtest_sole_header_test .............   Passed    0.12 sec
13/41 Test #13: gtest_premature_exit_test ..........   Passed    0.33 sec
14/41 Test  #1: gtest-death-test_test ..............   Passed    0.57 sec
15/41 Test #14: gtest-printers_test ................   Passed    0.33 sec
16/41 Test #19: gtest-test-part_test ...............   Passed    0.52 sec
17/41 Test #20: gtest_throw_on_failure_ex_test .....   Passed    0.52 sec
      Start 23: gtest-unittest-api_test
      Start 24: gtest-death-test_ex_nocatch_test
      Start 25: gtest-death-test_ex_catch_test
      Start 26: gtest_no_rtti_unittest
      Start 27: gtest-tuple_test
      Start 28: gtest_use_own_tuple_test
      Start 29: gtest_break_on_failure_unittest
18/41 Test #24: gtest-death-test_ex_nocatch_test ...   Passed    0.31 sec
19/41 Test #18: gtest_stress_test ..................   Passed    0.86 sec
20/41 Test #23: gtest-unittest-api_test ............   Passed    0.33 sec
21/41 Test #25: gtest-death-test_ex_catch_test .....   Passed    0.43 sec
22/41 Test #21: gtest-typed-test_test ..............   Passed    0.95 sec
23/41 Test #27: gtest-tuple_test ...................   Passed    0.53 sec
24/41 Test #28: gtest_use_own_tuple_test ...........   Passed    0.53 sec
      Start 30: gtest_catch_exceptions_test
      Start 31: gtest_color_test
      Start 32: gtest_env_var_test
      Start 33: gtest_filter_unittest
      Start 34: gtest_help_test
      Start 35: gtest_list_tests_unittest
      Start 36: gtest_output_test
25/41 Test #29: gtest_break_on_failure_unittest ....   Passed    0.84 sec
26/41 Test #32: gtest_env_var_test .................   Passed    0.30 sec
27/41 Test #30: gtest_catch_exceptions_test ........***Failed    0.32 sec
.FFFFFF.
======================================================================
FAIL: testCatchesCxxExceptionsInSetUp (__main__.CatchCxxExceptionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/googletest/test/gtest_catch_exceptions_test.py", line 176, in testCatchesCxxExceptionsInSetUp
    in EX_BINARY_OUTPUT)
AssertionError: False is not true

======================================================================
FAIL: testCatchesCxxExceptionsInSetUpTestCase (__main__.CatchCxxExceptionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/googletest/test/gtest_catch_exceptions_test.py", line 148, in testCatchesCxxExceptionsInSetUpTestCase
    in EX_BINARY_OUTPUT)
AssertionError: False is not true

======================================================================
FAIL: testCatchesCxxExceptionsInTearDown (__main__.CatchCxxExceptionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/googletest/test/gtest_catch_exceptions_test.py", line 194, in testCatchesCxxExceptionsInTearDown
    in EX_BINARY_OUTPUT)
AssertionError: False is not true

======================================================================
FAIL: testCatchesCxxExceptionsInTearDownTestCase (__main__.CatchCxxExceptionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/googletest/test/gtest_catch_exceptions_test.py", line 171, in testCatchesCxxExceptionsInTearDownTestCase
    in EX_BINARY_OUTPUT)
AssertionError: False is not true

======================================================================
FAIL: testCatchesCxxExceptionsInTestBody (__main__.CatchCxxExceptionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/googletest/test/gtest_catch_exceptions_test.py", line 205, in testCatchesCxxExceptionsInTestBody
    in EX_BINARY_OUTPUT)
AssertionError: False is not true

======================================================================
FAIL: testCatchesNonStdCxxExceptions (__main__.CatchCxxExceptionsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/googletest/test/gtest_catch_exceptions_test.py", line 218, in testCatchesNonStdCxxExceptions
    in EX_BINARY_OUTPUT)
AssertionError: False is not true

----------------------------------------------------------------------
Ran 8 tests in 0.005s

FAILED (failures=6)

28/41 Test #34: gtest_help_test ....................   Passed    0.49 sec
29/41 Test #35: gtest_list_tests_unittest ..........   Passed    0.48 sec
30/41 Test #33: gtest_filter_unittest ..............   Passed    0.61 sec
31/41 Test #31: gtest_color_test ...................   Passed    0.62 sec
32/41 Test #36: gtest_output_test ..................   Passed    0.57 sec
      Start 37: gtest_shuffle_test
      Start 38: gtest_throw_on_failure_test
      Start 39: gtest_uninitialized_test
      Start 40: gtest_xml_outfiles_test
      Start 41: gtest_xml_output_unittest
33/41 Test #37: gtest_shuffle_test .................   Passed    0.23 sec
34/41 Test #41: gtest_xml_output_unittest ..........   Passed    0.30 sec
35/41 Test #22: gtest_unittest .....................   Passed    2.01 sec
36/41 Test #40: gtest_xml_outfiles_test ............   Passed    0.32 sec
37/41 Test #39: gtest_uninitialized_test ...........   Passed    0.42 sec
38/41 Test #26: gtest_no_rtti_unittest .............   Passed    1.59 sec
39/41 Test #38: gtest_throw_on_failure_test ........   Passed    0.64 sec
40/41 Test #11: gtest-port_test ....................   Passed    2.60 sec
41/41 Test #16: gtest_repeat_test ..................   Passed    2.76 sec

98% tests passed, 1 tests failed out of 41

Total Test time (real) =   3.01 sec

The following tests FAILED:
     30 - gtest_catch_exceptions_test (Failed)
Errors while running CTest
Makefile:119: recipe for target 'test' failed
make[1]: *** [test] Error 8
```

--- a/googletest/src/gtest.cc
+++ b/googletest/src/gtest.cc
@@ -2675,10 +2675,12 @@ void TestInfo::Run() {
     test->Run();
   }
 
+  if (test != NULL) {
     // Deletes the test object.
     impl->os_stack_trace_getter()->UponLeavingGTest();
     internal::HandleExceptionsInMethodIfSupported(
-        test, &Test::DeleteSelf_, "the test fixture's destructor");
+	test, &Test::DeleteSelf_, "the test fixture's destructor");
+  }
 
   result_.set_elapsed_time(internal::GetTimeInMillis() - start);
 

